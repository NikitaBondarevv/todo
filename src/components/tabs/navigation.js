import PropTypes from 'prop-types'

import styles from './styles.css'
import { getDates } from '../../helpers/constans'

export const Navigation = ({ titles, setActiveTab, activeTabIndex }) => (
  <nav>
    <ul className={styles.list}>
      {titles.map((title, index) => (
        <li key={index} className={activeTabIndex === index ? styles.active : ''}>
          <a onClick={() => setActiveTab(index)}>
            {title}
            <br />
            <time className={styles.dates}>{getDates()[index]}</time>
          </a>
        </li>
      ))}
    </ul>
  </nav>
)

Navigation.defaultProps = {
  titles: [],
  activeTabIndex: 0,
  setActiveTab: () => { },
}

Navigation.propTypes = {
  titles: PropTypes.array,
  activeTabIndex: PropTypes.number,
  setActiveTab: PropTypes.func,
}
