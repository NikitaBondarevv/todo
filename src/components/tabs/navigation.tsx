import PropTypes from 'prop-types'

import styles from './styles.css'
import { getDates } from '../../helpers/constans'
import { TNavigationProps } from './types'

export const Navigation = ({ titles, setActiveTab, activeTabIndex }: TNavigationProps) => (
  <nav>
    <ul className={styles.list}>
      {titles.map((title, index) => (
        <li key={index} className={activeTabIndex === index ? styles.active : ''}>
          <a onClick={() => setActiveTab(index)}>
            {title}
            <br />
            <time className={styles.dates}>{getDates()[index]}</time>
          </a>
        </li>
      ))}
    </ul>
  </nav>
)

Navigation.defaultProps = {
  titles: [],
  activeTabIndex: 0,
}

Navigation.propTypes = {
  titles: PropTypes.array,
  activeTabIndex: PropTypes.number,
  setActiveTab: PropTypes.func,
}
