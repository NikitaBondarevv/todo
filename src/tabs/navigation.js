import styles from './styles.css'

export const Navigation = ({ titles, setActiveTab, activeTabIndex }) => {
  return (
    <nav>
      <ul className={styles.list} >
        {titles.map((title, index) =>
          <li key={index} className={activeTabIndex === index ? styles.active : ''} ><a href="#" onClick={() => setActiveTab(index)}>{title}</a></li>
        )}
      </ul>
    </nav>
  )
}