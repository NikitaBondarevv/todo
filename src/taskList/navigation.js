import styles from './styles.css'

import PropTypes from 'prop-types'

export const Navigation = ({ titles, setActiveTab, activeTabIndex }) => {
  return (
    <nav className={styles.tabsNav} >
      <ul className={styles.list} >
        {titles.map((title, index) =>
          <li key={index} className={activeTabIndex === index ? styles.active : ''} ><a href="#" onClick={() => setActiveTab(index)}>{title}</a></li>
        )}
      </ul>
    </nav>
  )
}

Navigation.defaultProps = {
  titles: [],
  activeTabIndex: 0,
  setActiveTab: function () {},
}

Navigation.propTypes = {
  titles: PropTypes.array,
  activeTabIndex: PropTypes.number,
  setActiveTab: PropTypes.func,
}
