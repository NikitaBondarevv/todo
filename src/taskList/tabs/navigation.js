import PropTypes from 'prop-types'

import styles from './styles.css'
import { dates } from '../../helpers/constans'

export const Navigation = ({ titles, setActiveTab, activeTabIndex }) => (
  <nav>
    <ul className={styles.list} >
      {titles.map((title, index) =>
        <li key={index} className={activeTabIndex === index ? styles.active : ''} >
          <a onClick={() => setActiveTab(index)}>
            {title}
            <br />
            <time className={styles.dates}>{dates[index]}</time>
          </a>
        </li>
      )}
    </ul>
  </nav>
)

Navigation.defaultProps = {
  titles: [],
  dates: [],
  activeTabIndex: 0,
  setActiveTab: () => { },
}

Navigation.propTypes = {
  titles: PropTypes.array,
  dates: PropTypes.array,
  activeTabIndex: PropTypes.number,
  setActiveTab: PropTypes.func,
}
