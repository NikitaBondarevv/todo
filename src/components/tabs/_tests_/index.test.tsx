import { render } from '@testing-library/react'

import { Tabs } from '..'
import { Navigation } from '../navigation'

jest.mock('components/tabs', () => ({
  Tabs: () => <div />
}))

jest.mock('components/tabs/navigation', () => ({
  Navigation: () => <nav />
}))

describe('<Tabs />, <Navigation />', () => {
  test('should render div', () => {
    const { asFragment } = render(
      <Tabs />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render nav', () => {
    const { asFragment } = render(
      <Navigation />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
