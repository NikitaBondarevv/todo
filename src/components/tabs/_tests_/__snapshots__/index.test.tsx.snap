// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs />, <Navigation /> should render div 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`<Tabs />, <Navigation /> should render nav 1`] = `
<DocumentFragment>
  <nav />
</DocumentFragment>
`;
